src/Xamarin.Android.Build.Tasks/Tasks/BuildArchive.cs

```diff
@@ -0,0 +1,253 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Framework;
+using Xamarin.Tools.Zip;
+
+namespace Xamarin.Android.Tasks;
+
+/// <summary>
+/// Takes a list of files and adds them to an APK archive. If the APK archive already
+/// exists, files are only added if they were changed.
+/// </summary>
+public class BuildArchive : AndroidTask
+{
+	public override string TaskPrefix => "BAA";
+
+	public string? AndroidPackageFormat { get; set; }
+
+	public string? ApkInputPath { get; set; }
+
+	[Required]
+	public string ApkOutputPath { get; set; } = null!; // NRT enforced by [Required]
+
+	[Required]
+	public ITaskItem [] FilesToAddToArchive { get; set; } = null!; // NRT enforced by [Required]
```

Comment:
Or:
```suggestion
	[Required]
	public ITaskItem [] FilesToAddToArchive { get; set; } = [];
```

Reply:
In general I avoid this pattern (and the empty string one mentioned: `= ""`), as it creates a new Array which is then overwritten by the array created when the Engine fills in the property, creating an extra unnecessary allocation and disposal.

Though maybe nowadays to compiler is smart enough to eliminate the unused objects?

Obviously in this case the difference of a few extra objects won't be a measurable difference, so I can change it if desired.

Reply:
`Array.Empty` would be better as its a shared array.

Reply:
`[]` should be the same as `Array.Empty()` now, I generally think either `[]` or `""` are prettier than `null!`. And I think both should be shared for the whole application?

Reply:
Updated